packages/plugin-ext/src/plugin/tree/tree-views.ts

```diff
@@ -159,6 +176,10 @@ class TreeViewExtImpl<T> implements Disposable {
     private readonly onDidChangeVisibilityEmitter = new Emitter<TreeViewVisibilityChangeEvent>();
     readonly onDidChangeVisibility = this.onDidChangeVisibilityEmitter.event;
 
+    protected updateMessage = debounce(() => {
```

Comment:
Why to debounce? Did you observe any performance or usability issues?

Reply:
At first invocation of `Find All References` or `Find All Implementations` command, the Reference View widget is not yet created, so I cannot set a message on it. So, when you ask for the references or implementation for the first time the references tree will be shown without that search summary message. Debouncing solves this problem.

Reply:
No, we cannot just "do it later". The problem here is that the state of the tree view can be changed before the widget in the UI is actually created. This just means that we need to transfer the state at the time the widget is created. But this can be 5 minutes after creation of the tree view, so the debounce() is not reliable. 
IMO, we have to pass message and title to TreeViewsMain#reveal(). I think that is the only way a plugin can make a tree view visible, right?

Reply:
Note sure where to put the message when the widget is not created yet. @akosyakov any suggestions how to do that?

Reply:
Had a discussion on Friday and will investigate what the behaviour of `createTreeView` should be: if it is supposed to open the widget, the `set message` should come afterwards and succeed.

Reply:
The widget creation happens in several steps and is distributed in several events and backend/frontend communications.
During this process `$setMessage` and `$setTitle` are invoked with default values `message=undefined` and `title="Results"` in order to indicate that the reference tree is empty during the search for the references (the results are not received yet) - trying to process this initial call is the reason of `debounce` appearance.
After the widget creation these methods are called once again with the real (not empty) values.

So, the simple way to get rid of debouncing is just the rejecting to set an empty message (doing nothing when `$setMessage` is invoked with `message=undefined` value). All the following calls to `$setMessage` will not be failed because the widget will be already created

Reply:
I think it should be part of tree view metadata which initially loaded from package.json. Then `$setMessage` or `$setTitle` are called when metadata should be updated and used during creation. If a view already present when we update it as well.